issues.go

```diff
@@ -192,3 +201,70 @@ func issueSearch(ctx context.Context, label, token string, ch chan<- Issue) erro
 	}
 	return nil
 }
+
+type languageFetcher struct {
+	fetchedRepos map[string][]string
+}
+
+func newLanguageFetcher() *languageFetcher {
+	return &languageFetcher{
+		fetchedRepos: make(map[string][]string),
+	}
+}
+
+func (lf *languageFetcher) repoLanguages(ctx context.Context, repoURL, token string) ([]string, error) {
+	// Return cached languages, if all ready fetched from repo.
+	if len(lf.fetchedRepos[repoURL]) > 0 {
+		return lf.fetchedRepos[repoURL], nil
+	}
+
+	// If not cached, get languages from repo.
+	req, err := http.NewRequest("GET", fmt.Sprintf("%v/languages", repoURL), nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not build request")
+	}
+
+	// Tell the request to use our context so we can cancel it in-flight if needed
+	req = req.WithContext(ctx)
+
+	// Use their access token so it counts against their rate limit
+	if token != "" {
+		req.Header.Add("Authorization", "token "+token)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not execute request")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return nil, errors.Wrapf(err, "status was %d, not 200", resp.StatusCode)
+	}
+	data := make(map[string]int)
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return nil, errors.Wrap(err, "could not decode json")
+	}
+
+	// Do golang limbo to get sorted languages.
```

Comment:
There is a potential problem here if two languages are present with the same number of bytes. It's pretty unlikely but it's worth thinking about. We could also move all of this limbo out to a function to make it easier to read like https://gist.github.com/jcbwlkr/2d9e8029defe63ca1330284dad37239f

Reply:
I was trying to come up with a good solution to make sure that doesn't happen. I haven't had any good ideas yet. Maybe, we should use a struct instead of a map?

Reply:
In the gist I linked the problem is solved. When we are parsing the map we put it into a `ranked := make(map[int][]string)`